manager/dispatcher/assignments.go

```diff
@@ -45,6 +48,29 @@ func assignSecret(a *assignmentSet, readTx store.ReadTx, mapKey typeAndID, t *ap
 		}).Debug("failed to fetch secret")
 		return
 	}
+	// If a driver is used, give a unique ID per task to allow different values for different tasks
+	if secret.Spec.Driver != nil {
+		// Give the secret a new ID and mark it as internal
+		originalSecretID := secret.ID
+		secret.ID = identity.NewID()
+		secret.Internal = true
+		// Create a new mapKey with the new ID and insert it into the dependencies map for the task.
+		// This will make the changes map contain an entry with the new ID rather than the original one.
+		mapKey = typeAndID{objType: mapKey.objType, id: secret.ID}
+		a.tasksUsingDependency[mapKey] = make(map[string]struct{})
+		a.tasksUsingDependency[mapKey][t.ID] = struct{}{}
+		// Update all references to the secret in the task's secrets or resource references to refer to the new ID
+		for _, ref := range t.Spec.ResourceReferences {
+			if ref.ResourceType == api.ResourceType_SECRET && ref.ResourceID == originalSecretID {
+				ref.ResourceID = secret.ID
```

Comment:
this is a big problem. the spec should never be written to by swarmkit, it should only be written by the user.

to put it one way, the spec is what the user says they want. swarmkit might choose to not do what the user wants (for example, if the user wants something that can't be done), but it can't ever re-write what the user wants.

Reply:
This should have been addressed now, with the `SecretMappings` field in the Task proto as a sibling to the `Spec` field.